src/rgen/Microsoft.Macios.Generator/Attributes/BindFromData.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Macios.Generator.Attributes;
+
+readonly struct BindFromData : IEquatable<BindFromData> {
+
+	public string Type { get; }
+	public string? OriginalType { get; }
+
+	public BindFromData (string type)
+	{
+		Type = type;
+	}
+
+	public BindFromData (string type, string? originalType)
```

Comment:
can't you juste use ITypeSymbols instead of strings? ToDisplayString() with the default SymbolFormater scares me a bit...

Reply:
+1 on being worried. It is a problem I have and I'm leaving it for later in the project. 

I want to minimize the size of the structure stored in memory and I want to be able to reduce the number of classes and simplify their tree to reduce the GC pressure during the compilation. Ideally, I would move to reference readonly structures but the API from roslyn does not allow that. 

Once we have the tests passing I will profile the generator with the strings and take a different approach. We do know that the bgen code had a HUGE memory usage due to strings (Up to 2Gb)